src/ser/mod.rs

```diff
@@ -741,6 +743,11 @@ mod tests {
 
     #[test]
     fn struct_() {
+        #[derive(Serialize)]
+        struct Nothing;
+
+        assert_eq!(to_string(&Nothing).unwrap(), r#"{}"#);
```

Comment:
Could you add a test that verifies this is serialized the same way as [serde_json](https://crates.io/crates/serde_json) does (for both `Nothing` and `Empty`)?

Reply:
OK, good idea.

Reply:
Not necessarily with this type, but we require this compatibility in the VM-contract communication. The VM uses serde_json (because it is faster and more mature) and the Rust contracts use serde-json-wasm. When the VM sends to the contract, JSON is serialized using serde_json and deserialized using serde-json-wasm. When the contract sends to the VM, vice-versa.

Reply:
In fact, I tested it and whereas `Empty` is serialized as `{}`, `Nothing` is serialized as `null`... Sorry, I thought the same lib was being used on both sides.

Will change the impl to serialize / deserialize as `null`. And, take a look at the `serde_json` code for reference, now that I think of it.

Reply:
No need to be sorry :) What we built there is a non-trivial stack of software and I'm happy to help you understand all the tricky bits (that are not documented anywhere).

Reply:
Fixed it. I still think that serializing unit structs to "{}" makes more sense than to "null". But better to be compatible.

Now, AFAIK `serde_json::Deserialize` doesn't support **deserialization** of "null" to unit stucts... maybe that's the right thing to do, to avoid potential errors / confusions.

Let me know what you think / prefer, and I can revert / reset those commits.